### ISSUE TITLE
UX430 (kabylake refresh) and realtek alc-294

### ISSUE THREAD

**Issue by xseignard** [hieplpvip/ASUS-ZENBOOK-UX410-HACKINTOSH#67]
Hello,
I have an Asus UX430 (kabylake refresh) and it seems that I have a realtek alc-294 instead of realtek alc-295 mentionned in this repository.

I had troubles getting the jack output working (but it was recognized when I plug my earphones).

I updated `AppleALC.kext` from here (https://github.com/acidanthera/AppleALC). And started fiddling to find the right `layout-id`.

At the end, `layout-id 21` is the one working for me.

So I created a file (`layout21_HDEF.asl` ) with the follwing (in `src/hotpatch/include`):
```
    // inject properties for audio

    External(_SB.PCI0.HDEF, DeviceObj)
    Method(_SB.PCI0.HDEF._DSM, 4)
    {
        If (!Arg2) { Return (Buffer() { 0x03 } ) }
        Return(Package()
        {
            "layout-id", Buffer(4) { 21, 0, 0, 0 },
            "hda-gfx", Buffer() { "onboard-1" },
            "PinConfigurations", Buffer() { },
        })
    }

```

And changed the `src/hotpatch/SSDT-UX430-KabyLakeR.dsl` to include the right layout ([on line 9](https://github.com/hieplpvip/ASUS-ZENBOOK-HACKINTOSH/blob/master/src/hotpatch/SSDT-UX430-KabyLakeR.dsl#L9)):
```
    #include "include/layout21_HDEF.asl"
```

Now I have working audio (jack + speakers)

I hope this help other people.
I can make a PR, but since it seems there are UX430 kabylake refresh with alc-295, I wouldn't mess with it.

Regards

**Comment by Vmonheros** [comment]
Hi.

I have an UX430 UAR with all-294 as you. And I also use layout 21 from the beginning without any issue (outside of the fact that there is no microphone with jack).

I inject Device Properties for codec configuration, one SSDT less :p.

<key>PciRoot(0)/Pci(0x1f,3)</key>
			<dict>
				<key>#no-controller-patch</key>
				<integer>1</integer>
				<key>hda-gfx</key>
				<string>onboard-1</string>
				<key>layout-id</key>
				<integer>21</integer>
			</dict>

**Comment by akaditya13** [comment]
Hello. I have UX430UNR with ALC294. To make my audio working, I updated AppleALC. Then on clover configurator I disabled patch that say rename HDAS to HDEF and change layout-id to 21. With this, I have jack audio. the microphone with jack is working too.

**Comment by Vmonheros** [comment]
Hi.

You have the microphone for your jack headset working ? Internal microphone works, but external headset one not in my case.

**Comment by akaditya13** [comment]
Yes, microphone on my headset is works too. i confirm it with trying alternately build in and the headset. But, my headset only working system preferences-sound-input window is active. so wierd.

**Comment by xseignard** [comment]
Hi @Vmonheros and @akaditya13 
Does the jack output work after sleep/wake ?

**Comment by bliss95** [comment]
Hi @xseignard 
The headset works properly for mic/out after sleep/wake.

I am using Multi system, Windows 10 and Hackintosh
My problem, no sound after restart Windows 10. To fix this, is to shutdown the Windows 10.

**Comment by xseignard** [comment]
Thanks @bliss95 but what's your asus model, realtek model, and your configuration ?

**Comment by bliss95** [comment]
https://www.asus.com/Laptops/ASUS-ZenBook-UX430UN/
i7 8550U, 16GB, 850GB Nvme, MX150 etc.

**Comment by xseignard** [comment]
Thanks, what about the realtek chip ?
Is it ALC294?
What layout id are you using? Do you also have codeccommander ?

**Comment by bliss95** [comment]
Realtek ALC294, Layout Id is 21, other layouts don't work, New Layout Id 22 is coming to test in AppleALC 1.4.9.
I have used CodecCommander before Catalina. Fixed the microphone issue.
I don't know if there is still a need now. It just stands in the EFI folder.

**Comment by xseignard** [comment]
Thanks a lot!

**Comment by bliss95** [comment]
AppleALC 1.4.9 is out.
Layout ID 22 is the same 21. No difference except Codec Name in plist

**Comment by yeboster** [comment]
Hello,

First of all thanks for this awesome work, I've got an Asus ux430unr and I've setup MacOs Mojave with little effort. What an experience MacOs!

How can I find what ALC I'm running ? And then, how can I update the correct layout ? I've read the [instructions of AppleALC](https://github.com/acidanthera/AppleALC/wiki/Installation-and-usage) and updated the kext but still the headphones don't work.

Thank you ahead.

---

**EDIT**: I've found I have the ALC294 as you. I've followed your instructions and now I'm a happy user of a fully functional mac :)
